manifests/load.pp

```diff
@@ -18,7 +18,16 @@
 
   case $ensure {
     'present': {
-      $changes = "clear '${name}'"
+      case $::osfamily {
+        'Debian': {
+          $changes = "clear '${name}'"
+        }
+        'Suse': {
+          $changes = "set MODULES_LOADED_ON_BOOT/value[last()+1] '${name}'"
+          $onlyif  = "match MODULES_LOADED_ON_BOOT/value[.='${name}'] not_include ${name}"
```

Comment:
You could make this idempotent by doing this:

```puppet
"set MODULES_LOADED_ON_BOOT/value[.='${name}'] '${name}'"
```

without an `onlyif` statement.